lib/src/main/java/com/shopify/checkoutsheetkit/errors/CheckoutErrorDecoder.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * MIT License
+ *
+ * Copyright 2023-present, Shopify Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.shopify.checkoutsheetkit.errors
+
+import com.shopify.checkoutsheetkit.LogWrapper
+import com.shopify.checkoutsheetkit.WebToSdkEvent
+import kotlinx.serialization.json.Json
+
+internal class CheckoutErrorDecoder @JvmOverloads constructor(
+    private val decoder: Json,
+    private val log: LogWrapper = LogWrapper()
+) {
+    fun decode(decodedMsg: WebToSdkEvent): CheckoutErrorPayload? = try {
+        val errors = decoder.decodeFromString<List<CheckoutErrorPayload>>(decodedMsg.body)
+        errors.firstOrNull()
```

Comment:
I think you'll want this to throw if there is no "first" event here so that the failed decoding is logged